client.go

```diff
@@ -323,8 +324,9 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 			return nil, nil, err
 		}
 	}
-
-	conn := newConn(netConn, false, d.ReadBufferSize, d.WriteBufferSize, d.WriteBufferPool, nil, nil, nil)
+	// set the reader to have a limit of 6KB to prevent a DOS via oversized headers
+	br := bufio.NewReaderSize(io.LimitReader(netConn, 6144), d.ReadBufferSize)
```

Comment:
Yeah I had it that way initially  but would have to rewrap the LimitReader in another bufio.Reader since that's what textProto expects. Or change the http.ReadResponse interface. 

Is either of those preferrable?

Reply:
Yeah which is why I reset it to a new bufferedReader below before returning. That's the part I wasn't necessarily sure about. It's either mutating the conn.br afterwards or doublewrapping it when calling `http.ReadResponse`